ciao-cli/image.go

```diff
@@ -362,19 +361,19 @@ func (cmd *imageDownloadCommand) run(args []string) error {
 		fatalf("Could not download image [%s]\n", err)
 	}
 
-	b, err := ioutil.ReadAll(r)
-	if err != nil {
-		fatalf("Could not read [%s]\n", err)
-	}
-
-	if cmd.file == "" {
-		fmt.Printf("%s\n", b)
-	} else {
-		err := ioutil.WriteFile(cmd.file, b, 0644)
+	dest := os.Stdout
+	if cmd.file != "" {
+		dest, err = os.Create(cmd.file)
```

Comment:
We're not closing the file we're creating.  I think we need to close it and check the err from close to be sure all the data is correctly written.